.github/workflows/Runtime-upgrade-reset.yaml

```diff
@@ -1,17 +1,50 @@
 name: Upgrade-runtime-reset
 
-on: [workflow_dispatch]
+on: workflow_dispatch
 
 env:
   CARGO_TERM_COLOR: always
+  NIGHTLY_TOOLCHAIN_VERSION: ${{ secrets.NIGHTLY_TOOLCHAIN_VERSION }}
+
 jobs:
   upgrade-runtime:
     runs-on: ubuntu-latest
     steps:
-      - name: Download wasm runtime
-        run: wget https://github.com/gear-tech/gear/releases/download/build/gear_runtime.compact.compressed.wasm
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Install nightly toolchain
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: nightly
+          components: clippy, rustfmt
+          target: wasm32-unknown-unknown
+
+      - name: Show specific nightly version
+        if: env.NIGHTLY_TOOLCHAIN_VERSION != ''
+        run: echo $NIGHTLY_TOOLCHAIN_VERSION | sed 's/-/ - /g'
+
+      - name: Install specific nightly toolchain
+        if: env.NIGHTLY_TOOLCHAIN_VERSION != ''
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: nightly-${{ env.NIGHTLY_TOOLCHAIN_VERSION }}
+          target: wasm32-unknown-unknown
+
+      - name: Install Protoc
+        uses: arduino/setup-protoc@v1
+
+      - name: Build binaries
+        run: |
+          cargo build -p gear-cli -p wasm-proc --profile production
```

Comment:
I don't see any reason to build node at all and wasm-proc via production profile.
Please, build wasm-proc with release flag and just simply single required runtime, without node.

Reply:
Agree that we are to build `gear-runtime` instead of `gear-cli`. Do you mean to build `gear-runtime` in the production profile and `wasm-proc` in the release one? I think it will take a longer time compared to building both in one profile.

Reply:
Agree. You can build only `gear-runtime` instead of binary.

Reply:
> Agree that we are to build `gear-runtime` instead of `gear-cli`. Do you mean to build `gear-runtime` in the production profile and `wasm-proc` in the release one?

Yeah, I meant exactly that.

> I think it will take a longer time compared to building both in one profile.

If I'm not mistaken, they have not so mane common deps to build. Production build takes 3-4x longer than release.

Reply:
> If I'm not mistaken, they have not so mane common deps to build. Production build takes 3-4x longer than release.

Ok, I'll compare them and choose the faster.

Reply:
Hard to understand while building on local M2 😄 But agree that `wasm-proc` doesn't have too many shared dependencies with `gear-runtime`, so I'll split them.

```
❯ cargo b -p gear-runtime -p wasm-proc --profile production
Finished production [optimized] target(s) in 2m 25s

❯ cargo b -p gear-runtime --profile production
Finished production [optimized] target(s) in 2m 11s

❯ cargo b -p wasm-proc --release
Finished release [optimized] target(s) in 18.44s
```

Reply:
> Hard to understand while building on local M2 😄 But agree that `wasm-proc` doesn't have too many shared dependencies with `gear-runtime`, so I'll split them.
> 
> ```
> ❯ cargo b -p gear-runtime -p wasm-proc --profile production
> Finished production [optimized] target(s) in 2m 25s
> 
> ❯ cargo b -p gear-runtime --profile production
> Finished production [optimized] target(s) in 2m 11s
> 
> ❯ cargo b -p wasm-proc --release
> Finished release [optimized] target(s) in 18.44s
> ```

Interesting results. Did you clean the caches between runs?

Reply:
Anyway, nice to hear that we came to consensus on separate builds 👍

Reply:
> Interesting results. Did you clean the caches between runs?

Yes, I've deleted the `target` directory before every build.

Reply:
Done.